wagtail/snippets/views/snippets.py

```diff
@@ -191,6 +191,20 @@ def get_list_filter(self):
             self.list_filter = self.list_filter(self.request)
         return super().get_list_filter()
 
+    def get_search_fields(self):
+        # Allow the search_fields to be a callable that takes a request
+        # so that it can be evaluated in the context of the request
```

Comment:
Thanks, I agree. I've changed it as such.

I've also updated the documentation of `search_fields` to note that it will only fall back to the model's `search_fields` attribute if a Wagtail search backend is used. If the backend is unset (to use the Django ORM), this fallback doesn't happen. If we want to replicate the fallback as well, then we'll need a `get_search_fields()` in the generic IndexView that does this.